hazelcast-sql/src/main/java/com/hazelcast/sql/impl/type/QueryDataTypeUtils.java

```diff
@@ -196,94 +182,15 @@ public final class QueryDataTypeUtils {
      */
     public static final MathContext DECIMAL_MATH_CONTEXT = new MathContext(MAX_DECIMAL_PRECISION, RoundingMode.HALF_UP);
 
-    private QueryDataTypeUtils() {
-        // No-op.
-    }
+    private QueryDataTypeUtils() { }
 
-    @SuppressWarnings({"checkstyle:CyclomaticComplexity", "checkstyle:ReturnCount", "checkstyle:MethodLength"})
     public static QueryDataType resolveTypeForClass(Class<?> clazz) {
         Converter converter = Converters.getConverter(clazz);
-
-        QueryDataTypeFamily typeFamily = converter.getTypeFamily();
-
-        switch (typeFamily) {
-            case VARCHAR:
-                if (converter == StringConverter.INSTANCE) {
-                    return VARCHAR;
-                } else {
-                    assert converter == CharacterConverter.INSTANCE;
-
-                    return VARCHAR_CHARACTER;
-                }
-
-            case BOOLEAN:
-                return BOOLEAN;
-
-            case TINYINT:
-                return TINYINT;
-
-            case SMALLINT:
-                return SMALLINT;
-
-            case INTEGER:
-                return INT;
-
-            case BIGINT:
-                return BIGINT;
-
-            case DECIMAL:
-                if (converter == BigDecimalConverter.INSTANCE) {
-                    return DECIMAL;
-                } else {
-                    assert converter == BigIntegerConverter.INSTANCE;
-
-                    return DECIMAL_BIG_INTEGER;
-                }
-
-            case REAL:
-                return REAL;
-
-            case DOUBLE:
-                return DOUBLE;
-
-            case DATE:
-                return DATE;
-
-            case TIME:
-                return TIME;
-
-            case TIMESTAMP:
-                return TIMESTAMP;
-
-            case TIMESTAMP_WITH_TIME_ZONE:
-                if (converter == DateConverter.INSTANCE) {
-                    return TIMESTAMP_WITH_TZ_DATE;
-                } else if (converter == CalendarConverter.INSTANCE) {
-                    return TIMESTAMP_WITH_TZ_CALENDAR;
-                } else if (converter == InstantConverter.INSTANCE) {
-                    return TIMESTAMP_WITH_TZ_INSTANT;
-                } else if (converter == OffsetDateTimeConverter.INSTANCE) {
-                    return TIMESTAMP_WITH_TZ_OFFSET_DATE_TIME;
-                } else {
-                    assert converter == ZonedDateTimeConverter.INSTANCE;
-
-                    return TIMESTAMP_WITH_TZ_ZONED_DATE_TIME;
-                }
-
-            case OBJECT:
-            case MAP:
-                return OBJECT;
-
-            case NULL:
-                return NULL;
-
-            case JSON:
-                return JSON;
-            case ROW:
-                return ROW;
-            default:
-                throw new IllegalArgumentException("Unexpected class: " + clazz);
+        QueryDataType type = QueryDataType.TYPES.get(converter);
+        if (type == null) {
```

Comment:
we discussed this already that converting switch with enum to map might decrease performance, so I would avoid doing that unless there are clear benefits.

Reply:
I also won't consider this approach with map. It just eats more memory with no execution time improvements.

Reply:
Good candidate to refactor with pattern matching in the future :)

Reply:
Moved to the upcoming PR.